include/aie/Dialect/AIE/IR/AIETargetModel.h

```diff
@@ -236,6 +236,9 @@ class AIETargetModel {
   // There are several special cases for handling the NPU at the moment.
   virtual bool isNPU() const { return false; }
 
+  // Return true if this device is using semaphore locks.
+  virtual bool isUsingSemaphoreLocks() const { return false; }
```

Comment:
There aren't any `hasProperty()` functions in `AIETargetModel` but this seems like it could be one. "is using" is a bit awkward to me.

Reply:
Good point. Method is reimplemented with `hasProperty(UsesSemaphoreLocks)`. Thanks for the pointer.